File: compatibility-tests/src/test/java/nnl/rocks/kactoos/scalar/RetryScalarTest.java
/**
 * The MIT License (MIT)
 * <p>
 * Copyright (c) 2017-2018 Yegor Bugayenko
 * <p>
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 * <p>
 * The above copyright notice and this permission notice shall be included
 * in all copies or substantial portions of the Software.
 * <p>
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */
package nnl.rocks.kactoos.scalar;

import org.hamcrest.MatcherAssert;
import org.hamcrest.Matchers;
import org.junit.Test;

import java.security.SecureRandom;

/**
 * Test case for {@link RetryScalar}.
 *
 * @author Yegor Bugayenko (yegor256@gmail.com)
 * @version $Id: 738e7b7210b77d4f2ccadc7cdc1d9cdcc5b4651e $
 * @since 0.9
 * @checkstyle JavadocMethodCheck (500 lines)
 */
public final class RetryScalarTest {

    @Test
    public void runsScalarMultipleTimes() throws Exception {
        MatcherAssert.assertThat(
            new RetryScalar<>(
                () -> {
                    // @checkstyle MagicNumberCheck (1 line)
                    if (new SecureRandom().nextDouble() > 0.3d) {
                        throw new IllegalArgumentException("May happen");
                    }
                    return 0;
                },
                Integer.MAX_VALUE
            ).value(),
            Matchers.equalTo(0)
        );
    }

}

File: kactoos-jvm/src/test/kotlin/nnl/rocks/kactoos/scalar/RetryScalarTest.kt

package nnl.rocks.kactoos.scalar

import org.hamcrest.MatcherAssert
import org.hamcrest.Matchers
import org.junit.Test

import java.security.SecureRandom

/**
 * Test case for [RetryScalar].
 *
 * @author Yegor Bugayenko (yegor256@gmail.com)
 * @version $Id: 738e7b7210b77d4f2ccadc7cdc1d9cdcc5b4651e $
 * @since 0.9
 * @checkstyle JavadocMethodCheck (500 lines)
 */
class RetryScalarTest {

    @Test
    @Throws(Exception::class)
    fun runsScalarMultipleTimes() {
        MatcherAssert.assertThat(
                RetryScalar<Int>(
                        {
                            // @checkstyle MagicNumberCheck (1 line)
                            if (SecureRandom().nextDouble() > 0.3) {
                                throw IllegalArgumentException("May happen")
                            }
                            0
                        },
                        Integer.MAX_VALUE
                ).value(),
                Matchers.equalTo(0)
        )
    }
}
